sql/parser/grant.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Marc Berhault (marc@cockroachlabs.com)
+
+// This code was derived from https://github.com/youtube/vitess.
+//
+// Copyright 2012, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file
+
+package parser
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Grant represents a GRANT statement.
+type Grant struct {
+	Privileges []string
+	Targets    TargetList
+	Grantees   []string
+}
+
+type TargetType int
+
+const (
+	TargetDatabase TargetType = iota
+)
+
+var targetNames = map[TargetType]string{
+	TargetDatabase: "DATABASE",
+}
+
+// TargetList represents a list of targets of a given type.
+type TargetList struct {
+	Type    TargetType
+	Targets []string
+}
+
+func (node *Grant) String() string {
+	return fmt.Sprintf("GRANT %s ON %s %s TO %v",
+		strings.Join(node.Privileges, ","),
```

Comment:
Oh, good point. Yeah, I think it might be best to make the privileges some sort of enumerated type like `TargetType`. But leave that for another PR. This looks good for now.